number.go

```diff
@@ -618,3 +620,376 @@ func (n *Number) Le(value interface{}) *Number {
+			return n
+		}
+	}
+
+	return n
+}
+
+// NotInt succeeds if number is not a signed integer of the specified bit
+// width as an optional argument.
+//
+// Bits argument defines maximum allowed bitness for the given number.
+// If bits is omitted, boundary checks is omitted too.
+//
+// value should have numeric type convertible to float64. Before comparison,
+// it is converted to float64.
+//
+// Example:
+//
+//	number := NewNumber(t, 123.0123)
+//	number.NotInt(32)
+//	number.NotInt(64)
+//	number.NotInt()
+func (n *Number) NotInt(bits ...int) *Number {
+	opChain := n.chain.enter("NotInt()")
+	defer opChain.leave()
+
+	if opChain.failed() {
+		return n
+	}
+
+	if len(bits) > 1 {
+		opChain.fail(AssertionFailure{
+			Type: AssertUsage,
+			Errors: []error{
+				errors.New("unexpected multiple bits arguments"),
+			},
+		})
+		return n
+	}
+
+	if !math.IsNaN(n.value) {
+		inum, acc := big.NewFloat(n.value).Int(nil)
+		if acc == big.Exact {
+			if len(bits) == 0 {
+				opChain.fail(AssertionFailure{
+					Type:   AssertType,
+					Actual: &AssertionValue{n.value},
+					Errors: []error{
+						errors.New("expected: number is not integer"),
+					},
+				})
+				return n
+			}
+
+			bitSize := bits[0]
+			imax := new(big.Int)
+			imax.Lsh(big.NewInt(1), uint(bitSize-1))
+			imax.Sub(imax, big.NewInt(1))
+			imin := new(big.Int)
+			imin.Neg(imax)
+			imin.Sub(imin, big.NewInt(1))
+			if !(inum.Cmp(imin) < 0 || inum.Cmp(imax) > 0) {
+				opChain.fail(AssertionFailure{
+					Type:   AssertNotInRange,
+					Actual: &AssertionValue{n.value},
+					Expected: &AssertionValue{AssertionRange{
+						fmt.Sprintf("-2^%d", bitSize-1),
+						fmt.Sprintf("2^%d", bitSize-1),
+					}},
+					Errors: []error{
+						fmt.Errorf("expected: number doesn't fit %d-bit integer", bitSize),
+					},
+				})
+				return n
+			}
+		}
+	}
+
+	return n
+}
+
+// IsUint succeeds if number is an unsigned integer of the specified bit
+// width as an optional argument.
+//
+// Bits argument defines maximum allowed bitness for the given number.
+// If bits is omitted, boundary checks is omitted too.
+//
+// value should have numeric type convertible to float64. Before comparison,
+// it is converted to float64.
+//
+// Example:
+//
+//	number := NewNumber(t, 123)
+//	number.IsUint(32)
+//	number.IsUint(64)
+//	number.IsUint()
+func (n *Number) IsUint(bits ...int) *Number {
+	opChain := n.chain.enter("IsUint()")
+	defer opChain.leave()
+
+	if opChain.failed() {
+		return n
+	}
+
+	if len(bits) > 1 {
+		opChain.fail(AssertionFailure{
+			Type: AssertUsage,
+			Errors: []error{
+				errors.New("unexpected multiple bits arguments"),
+			},
+		})
+		return n
+	}
+
+	if math.IsNaN(n.value) {
+		opChain.fail(AssertionFailure{
+			Type:   AssertType,
+			Actual: &AssertionValue{n.value},
+			Errors: []error{
+				errors.New("expected: number is unsigned integer"),
+			},
+		})
+		return n
+	}
+
+	inum, acc := big.NewFloat(n.value).Int(nil)
+	if !(acc == big.Exact && inum.IsUint64()) {
```

Comment:
IsUint64() check forbids larger integers, but we want to handle them. We should remove this check.

Instead, when bits is omitted, we should check that value is non-negative (because we want *unsigned* integer).

Reply:
Same for NotUint